beets/mediafile.py

```diff
@@ -153,21 +153,13 @@ def _safe_cast(out_type, val):
             return unicode(val)
 
     elif out_type == float:
-        if isinstance(val, int) or isinstance(val, float):
-            return float(val)
-        else:
-            if not isinstance(val, basestring):
+        try:
+            if not isinstance(val, (int, float, basestring)):
                 val = unicode(val)
-            match = re.match(r'[\+-]?[0-9\.]+', val.strip())
-            if match:
-                val = match.group(0)
```

Comment:
The regex is actually useful because it lets us parse strings that have numbers followed by non-number cruft. This is surprisingly common in tags, and just plain `float()` doesn't cut it in this case.